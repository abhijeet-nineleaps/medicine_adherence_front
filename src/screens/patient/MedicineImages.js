import {useFocusEffect} from '@react-navigation/native';
import React, {useState} from 'react';
import {
  Text,
  Dimensions,
  View,
  Image,
  FlatList,
  ActivityIndicator,
  LogBox,
} from 'react-native';
import { useRoute } from '@react-navigation/native';
import Carousel, {Pagination} from 'react-native-snap-carousel';
import {API_URL} from '../../repositories/var';
import styles from './patientStyles/MedicineImagesStyles';
import Logger from '../../components/logger';

LogBox.ignoreAllLogs();
const SLIDER_WIDTH = Dimensions.get('window').width;
const ITEM_WIDTH = Math.round(SLIDER_WIDTH * 0.84);
const CarouselCardItem = ({item}) => {
  const [load, setload] = useState(true);
 Logger.loggerInfo(item);
  return (
    <View style={styles.container1}>
      <Image
        source={{
          uri: `${API_URL}/upload/static/images/${item.imageUrl}`,
        }}
        style={styles.image}
        onLoadStart={() => setload(true)}
        onLoadEnd={() => setload(false)}
      />
      {load && <ActivityIndicator />}

      <View style={styles.bodyView}>
        <Text style={styles.body}>Medicine taken time:</Text>
        <Text style={styles.body1}>{item.time}</Text>
      </View>
      <View style={styles.bodyView1}>
        <Text style={styles.body}>Sent to - </Text>
        <Text style={styles.body1}>{item.caretakerName}</Text>
      </View>
    </View>
  );
};

const SingleImageComponent = ({item}) => {
 Logger.loggerInfo('item', item);
  const [index, setindex] = React.useState(0);

  return (
    <>
      <View style={styles.container}>
        <Text style={styles.date}>Date - {item[0].date}</Text>
      </View>
      <View style={styles.carousel}>
        <Carousel
          layoutCardOffset={9}
          data={item}
          onSnapToItem={inde => setindex(inde)}
          renderItem={({item}) => <CarouselCardItem item={item} />}
          sliderWidth={SLIDER_WIDTH}
          itemWidth={ITEM_WIDTH}
          inactiveSlideShift={0}
          useScrollView={true}
        />
        <Pagination
          dotsLength={item.length}
          activeDotIndex={index}
          containerStyle={styles.pageContainer}
          dotStyle={styles.pageDot}
          inactiveDotStyle={styles.inactiveDot}
          inactiveDotOpacity={0.4}
          inactiveDotScale={0.6}
        />
      </View>
    </>
  );
};

const MedicineImages = ({}) => {
  const [imageData, setImageData] = useState([]);
  const route=useRoute();
  const medId = route.params;
  function fetchImages() {
    fetch(`${API_URL}/api/v1/medicine-images?medId=${medId}`)
      .then(resp => resp.json())
      .then(response => {
        let map = new Map();
        for (let re = 0; re < response.length; re++) {
          if (map.has(response[re].date)) {
            map.get(response[re].date).push(response[re]);
          } else {
            let newarr = [];
            newarr.push(response[re]);
            map.set(response[re].date, newarr);
          }
        }
        let Arr = [];
        map.forEach(it => Arr.push(it));
        setImageData(Arr);
      });
  }
  useFocusEffect(
    React.useCallback(() => {
      fetchImages();
      return () => {
        /* do nothing */
      };
    }, []),
  );

  return (
    <View style={styles.top}>
      {imageData.length === 0 ? (
        <View style={styles.imgView}>
          <Image
            source={require('../../../assests/images/noImages.png')}
            style={styles.img}
            resizeMode="contain"></Image>
        </View>
      ) : (
        <FlatList
          data={imageData}
          renderItem={({item}) => <SingleImageComponent item={item} />}
        />
      )}
    </View>
  );
};
export default MedicineImages;