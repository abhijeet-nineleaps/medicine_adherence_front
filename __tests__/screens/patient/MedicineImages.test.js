import React from 'react';
import renderer from 'react-test-renderer';
import MedicineImages from '../../../src/screens/patient/MedicineImages';
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useFocusEffect: jest.fn(),
  useNavigation: () => ({ goBack: jest.fn() }),
  useRoute: () => ({
    params: {
     medId: {}
    }
  }),
}));
describe('Click send image', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MedicineImages/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
