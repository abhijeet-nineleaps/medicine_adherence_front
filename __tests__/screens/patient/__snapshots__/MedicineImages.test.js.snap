// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "position": "absolute",
        "top": 40,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assests/images/noImages.png",
        }
      }
      style={
        Object {
          "width": 250,
        }
      }
    />
  </View>
</View>
`;
